src/test/java/ser/mil/rideapp/domain/service/RideServiceTest.java

```diff
@@ -0,0 +1,141 @@
+
+        when(rideRepository.pendingRides()).thenReturn(new ArrayList<>(List.of(ride)));
+        when(rideRepository.availableDrivers()).thenReturn(new ArrayList<>(List.of(driver)));
+
+        //When
+        rideService.pairPassengerWithDriver();
+
+        //Then
+        verify(ride, times(1)).setDriver(driver);
+        verify(ride, times(1)).setStatus(RideStatus.PENDING);
+
+        verify(driver,times(1)).setAvailable(false);
+    }
+
+    @Test
+    void pairPassengerWithDriver_twoRides_oneDriver() {
+        //Given
+        Ride ride1 = spy(new Ride("1", "Lublin", "Krakow", "Kuba", 50, RideStatus.PENDING));
+        Ride ride2 = spy(new Ride("2", "Lublin", "Krakow", "Kuba", 50, RideStatus.PENDING));
+        Driver driver = spy(new Driver("1", "Kuba", "Matejczyk"));
+
+        when(rideRepository.pendingRides()).thenReturn(new ArrayList<>(List.of(ride1, ride2)));
+        when(rideRepository.availableDrivers()).thenReturn(new ArrayList<>(List.of(driver)));
+
+        //When
+        rideService.pairPassengerWithDriver();
+
+        //Then
+        verify(ride1, times(1)).setDriver(driver);
+        verify(ride2, never()).setDriver(driver);
+
+        verify(ride1, times(1)).setStatus(RideStatus.PENDING);
+        verify(ride2, never()).setStatus(RideStatus.PENDING);
+
+        verify(driver,times(1)).setAvailable(false);
+    }
+
+    @Test
+    void pairPassengerWithDriver_twoRides_threeDrivers() {
+        //Given
+        Ride ride1 = spy(new Ride("1", "Lublin", "Krakow", "Kuba", 50, RideStatus.PENDING));
+        Ride ride2 = spy(new Ride("2", "Lublin", "Krakow", "Kuba", 50, RideStatus.PENDING));
+        Driver driver1 = spy(new Driver("1", "Kuba", "Matejczyk"));
+        Driver driver2 = spy(new Driver("2", "Kuba", "Matejczyk"));
+        Driver driver3 = spy(new Driver("3", "Kuba", "Matejczyk"));
+
+        when(rideRepository.pendingRides()).thenReturn(new ArrayList<>(List.of(ride1, ride2)));
+        when(rideRepository.availableDrivers()).thenReturn(new ArrayList<>(List.of(driver1, driver2, driver3)));
+
+        //When
+        rideService.pairPassengerWithDriver();
+
+        //Then
+        verify(ride1, times(1)).setDriver(driver1);
+        verify(ride2, times(1)).setDriver(driver2);
+
+        verify(ride1, times(1)).setStatus(RideStatus.PENDING);
+        verify(ride2, times(1)).setStatus(RideStatus.PENDING);
+
+        verify(driver1, times(1)).setAvailable(false);
+        verify(driver2, times(1)).setAvailable(false);
+        verify(driver3, never()).setAvailable(false);
+    }
+
+    @Test
+    void pairPassengerWithDriver_oneRide_zeroDriver() {
+        //Given
+        Ride ride1 = spy(new Ride("1", "Lublin", "Krakow", "Kuba", 50, RideStatus.PENDING));
+        when(rideRepository.pendingRides()).thenReturn(new ArrayList<>(List.of(ride1)));
+
+        //When
+        rideService.pairPassengerWithDriver();
+
+        //Then
+        verify(ride1, never()).setDriver(null);
```

Comment:
Tutaj zamiast null sprobuj przekazac any() - null konkretnie precyzuje czego ride1 nie moze ustawic jako setDriver, ale moze byc czyms innym niz null i to tez nam sie nie powinno spodobac (bo lista jest pusta) - dlatego dajemy any() - ze niewazne co on ustawi, nie powinien tego ustawiac